[
    {
        "url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/22",
        "repository_url": "https://api.github.com/repos/rustbaeceans/discordterm",
        "labels_url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/22/labels{/name}",
        "comments_url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/22/comments",
        "events_url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/22/events",
        "html_url": "https://github.com/rustbaeceans/discordterm/issues/22",
        "id": 476395002,
        "node_id": "MDU6SXNzdWU0NzYzOTUwMDI=",
        "number": 22,
        "title": "Failed to run custom build command for `openssl-sys-extras v0.7.14`",
        "user": {
            "login": "avanasear",
            "id": 47052410,
            "node_id": "MDQ6VXNlcjQ3MDUyNDEw",
            "avatar_url": "https://avatars0.githubusercontent.com/u/47052410?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/avanasear",
            "html_url": "https://github.com/avanasear",
            "followers_url": "https://api.github.com/users/avanasear/followers",
            "following_url": "https://api.github.com/users/avanasear/following{/other_user}",
            "gists_url": "https://api.github.com/users/avanasear/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/avanasear/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/avanasear/subscriptions",
            "organizations_url": "https://api.github.com/users/avanasear/orgs",
            "repos_url": "https://api.github.com/users/avanasear/repos",
            "events_url": "https://api.github.com/users/avanasear/events{/privacy}",
            "received_events_url": "https://api.github.com/users/avanasear/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [],
        "state": "open",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 0,
        "created_at": "2019-08-03T00:13:29Z",
        "updated_at": "2019-08-03T00:13:29Z",
        "closed_at": null,
        "author_association": "NONE",
        "body": "I ran `RUST_BACKTRACE=1 cargo run` and this was the output:\r\n```\r\n   Compiling openssl-sys-extras v0.7.14\r\n   Compiling openssl v0.7.14\r\n   Compiling rand v0.3.23\r\n   Compiling tui v0.2.3\r\n   Compiling hpack v0.2.0\r\n   Compiling mime v0.2.6\r\n   Compiling num-integer v0.1.41\r\n   Compiling syn v0.15.42\r\n   Compiling sodiumoxide v0.0.12\r\nerror: failed to run custom build command for `openssl-sys-extras v0.7.14`\r\n\r\nCaused by:\r\n  process didn't exit successfully: `/home/alucard/builds/discordterm/target/debug/build/openssl-sys-extras-569bce7032713ed7/build-script-build` (exit code: 101)\r\n--- stdout\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nOPT_LEVEL = Some(\"0\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nCC_x86_64-unknown-linux-gnu = None\r\nCC_x86_64_unknown_linux_gnu = None\r\nHOST_CC = None\r\nCC = None\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nCFLAGS_x86_64-unknown-linux-gnu = None\r\nCFLAGS_x86_64_unknown_linux_gnu = None\r\nHOST_CFLAGS = None\r\nCFLAGS = None\r\nDEBUG = Some(\"true\")\r\nrunning: \"cc\" \"-O0\" \"-ffunction-sections\" \"-fdata-sections\" \"-fPIC\" \"-g\" \"-m64\" \"-Wall\" \"-Wextra\" \"-o\" \"/home/alucard/builds/discordterm/target/debug/build/openssl-sys-extras-28b87e3d3c3ef277/out/src/openssl_shim.o\" \"-c\" \"src/openssl_shim.c\"\r\ncargo:warning=src/openssl_shim.c: In function \u2018SSL_CTX_set_ecdh_auto_shim\u2019:\r\ncargo:warning=src/openssl_shim.c:121:41: warning: unused parameter \u2018ctx\u2019 [-Wunused-parameter]\r\ncargo:warning=  121 | int SSL_CTX_set_ecdh_auto_shim(SSL_CTX *ctx, int onoff) {\r\ncargo:warning=      |                                ~~~~~~~~~^~~\r\ncargo:warning=src/openssl_shim.c: In function \u2018DH_new_from_params\u2019:\r\ncargo:warning=src/openssl_shim.c:132:7: error: dereferencing pointer to incomplete type \u2018DH\u2019 {aka \u2018struct dh_st\u2019}\r\ncargo:warning=  132 |     dh->p = p;\r\ncargo:warning=      |       ^~\r\ncargo:warning=src/openssl_shim.c: In function \u2018X509_get_extensions_shim\u2019:\r\ncargo:warning=src/openssl_shim.c:143:13: error: dereferencing pointer to incomplete type \u2018X509\u2019 {aka \u2018struct x509_st\u2019}\r\ncargo:warning=  143 |     return x->cert_info ? x->cert_info->extensions : NULL;\r\ncargo:warning=      |             ^~\r\ncargo:warning=src/openssl_shim.c:144:1: warning: control reaches end of non-void function [-Wreturn-type]\r\ncargo:warning=  144 | }\r\ncargo:warning=      | ^\r\nexit code: 1\r\n\r\n--- stderr\r\nthread 'main' panicked at '\r\n\r\nInternal error occurred: Command \"cc\" \"-O0\" \"-ffunction-sections\" \"-fdata-sections\" \"-fPIC\" \"-g\" \"-m64\" \"-Wall\"\r\n\"-Wextra\" \"-o\" \"/home/alucard/builds/discordterm/target/debug/build/openssl-sys-extras-28b87e3d3c3ef277/out/src/openssl_shim.o\" \"-c\" \"src/openssl_shim.c\" with args \"cc\" did not execute successfully (status code exit code: 1).\r\n\r\n', /home/alucard/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:1672:5\r\nstack backtrace:\r\n   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace\r\n   1: std::sys_common::backtrace::_print\r\n   2: std::panicking::default_hook::{{closure}}\r\n   3: std::panicking::default_hook\r\n   4: std::panicking::rust_panic_with_hook\r\n   5: std::panicking::continue_panic_fmt\r\n   6: std::panicking::begin_panic_fmt\r\n   7: gcc::fail\r\n             at /home/alucard/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:1672\r\n   8: gcc::Build::compile\r\n             at /home/alucard/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:785\r\n   9: build_script_build::main\r\n             at ./build.rs:19\r\n  10: std::rt::lang_start::{{closure}}\r\n             at /build/rust/src/rustc-1.36.0-src/src/libstd/rt.rs:64\r\n  11: std::panicking::try::do_call\r\n  12: __rust_maybe_catch_panic\r\n  13: std::rt::lang_start_internal\r\n  14: std::rt::lang_start\r\n             at /build/rust/src/rustc-1.36.0-src/src/libstd/rt.rs:64\r\n  15: main\r\n  16: __libc_start_main\r\n  17: _start\r\n\r\nwarning: build failed, waiting for other jobs to finish...\r\nerror: failed to run custom build command for `openssl v0.7.14`\r\n\r\nCaused by:\r\n  process didn't exit successfully: `/home/alucard/builds/discordterm/target/debug/build/openssl-6303d2aded0df56f/build-script-build` (exit code: 101)\r\n--- stdout\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nOPT_LEVEL = Some(\"0\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nCC_x86_64-unknown-linux-gnu = None\r\nCC_x86_64_unknown_linux_gnu = None\r\nHOST_CC = None\r\nCC = None\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nTARGET = Some(\"x86_64-unknown-linux-gnu\")\r\nHOST = Some(\"x86_64-unknown-linux-gnu\")\r\nCFLAGS_x86_64-unknown-linux-gnu = None\r\nCFLAGS_x86_64_unknown_linux_gnu = None\r\nHOST_CFLAGS = None\r\nCFLAGS = None\r\nDEBUG = Some(\"true\")\r\nrunning: \"cc\" \"-O0\" \"-ffunction-sections\" \"-fdata-sections\" \"-fPIC\" \"-g\" \"-m64\" \"-Wall\" \"-Wextra\" \"-o\" \"/home/alucard/builds/discordterm/target/debug/build/openssl-bf45dad9f562eadf/out/src/c_helpers.o\" \"-c\" \"src/c_helpers.c\"\r\ncargo:warning=src/c_helpers.c: In function \u2018rust_SSL_clone\u2019:\r\ncargo:warning=src/c_helpers.c:4:5: warning: implicit declaration of function \u2018CRYPTO_add\u2019 [-Wimplicit-function-declaration]\r\ncargo:warning=    4 |     CRYPTO_add(&ssl->references, 1, CRYPTO_LOCK_SSL);\r\ncargo:warning=      |     ^~~~~~~~~~\r\ncargo:warning=src/c_helpers.c:4:20: error: dereferencing pointer to incomplete type \u2018SSL\u2019 {aka \u2018struct ssl_st\u2019}\r\ncargo:warning=    4 |     CRYPTO_add(&ssl->references, 1, CRYPTO_LOCK_SSL);\r\ncargo:warning=      |                    ^~\r\ncargo:warning=src/c_helpers.c:4:37: error: \u2018CRYPTO_LOCK_SSL\u2019 undeclared (first use in this function); did you mean \u2018CRYPTO_LOCK\u2019?\r\ncargo:warning=    4 |     CRYPTO_add(&ssl->references, 1, CRYPTO_LOCK_SSL);\r\ncargo:warning=      |                                     ^~~~~~~~~~~~~~~\r\ncargo:warning=      |                                     CRYPTO_LOCK\r\ncargo:warning=src/c_helpers.c:4:37: note: each undeclared identifier is reported only once for each function it\r\nappears in\r\ncargo:warning=src/c_helpers.c: In function \u2018rust_SSL_CTX_clone\u2019:\r\ncargo:warning=src/c_helpers.c:8:20: error: dereferencing pointer to incomplete type \u2018SSL_CTX\u2019 {aka \u2018struct ssl_ctx_st\u2019}\r\ncargo:warning=    8 |     CRYPTO_add(&ctx->references,1,CRYPTO_LOCK_SSL_CTX);\r\ncargo:warning=      |                    ^~\r\ncargo:warning=src/c_helpers.c:8:35: error: \u2018CRYPTO_LOCK_SSL_CTX\u2019 undeclared (first use in this function); did you mean \u2018CRYPTO_EX_INDEX_SSL_CTX\u2019?\r\ncargo:warning=    8 |     CRYPTO_add(&ctx->references,1,CRYPTO_LOCK_SSL_CTX);\r\ncargo:warning=      |                                   ^~~~~~~~~~~~~~~~~~~\r\ncargo:warning=      |                                   CRYPTO_EX_INDEX_SSL_CTX\r\ncargo:warning=src/c_helpers.c: In function \u2018rust_X509_clone\u2019:\r\ncargo:warning=src/c_helpers.c:12:21: error: dereferencing pointer to incomplete type \u2018X509\u2019 {aka \u2018struct x509_st\u2019}\r\ncargo:warning=   12 |     CRYPTO_add(&x509->references,1,CRYPTO_LOCK_X509);\r\ncargo:warning=      |                     ^~\r\ncargo:warning=src/c_helpers.c:12:36: error: \u2018CRYPTO_LOCK_X509\u2019 undeclared (first use in this function); did you\r\nmean \u2018CRYPTO_LOCK\u2019?\r\ncargo:warning=   12 |     CRYPTO_add(&x509->references,1,CRYPTO_LOCK_X509);\r\ncargo:warning=      |                                    ^~~~~~~~~~~~~~~~\r\ncargo:warning=      |                                    CRYPTO_LOCK\r\nexit code: 1\r\n\r\n--- stderr\r\nthread 'main' panicked at '\r\n\r\n\r\nInternal error occurred: Command \"cc\" \"-O0\" \"-ffunction-sections\" \"-fdata-sections\" \"-fPIC\" \"-g\" \"-m64\" \"-Wall\"\r\n\"-Wextra\" \"-o\" \"/home/alucard/builds/discordterm/target/debug/build/openssl-bf45dad9f562eadf/out/src/c_helpers.o\" \"-c\" \"src/c_helpers.c\" with args \"cc\" did not execute successfully (status code exit code: 1).\r\n\r\n', /home/alucard/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:1672:5\r\nstack backtrace:\r\n   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace\r\n   1: std::sys_common::backtrace::_print\r\n   2: std::panicking::default_hook::{{closure}}\r\n   3: std::panicking::default_hook\r\n   4: std::panicking::rust_panic_with_hook\r\n   5: std::panicking::continue_panic_fmt\r\n   6: std::panicking::begin_panic_fmt\r\n   7: gcc::fail\r\n             at /home/alucard/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:1672\r\n   8: gcc::Build::compile\r\n             at /home/alucard/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.55/src/lib.rs:785\r\n   9: build_script_build::main\r\n             at ./build.rs:15\r\n  10: std::rt::lang_start::{{closure}}\r\n             at /build/rust/src/rustc-1.36.0-src/src/libstd/rt.rs:64\r\n  11: std::panicking::try::do_call\r\n  12: __rust_maybe_catch_panic\r\n  13: std::rt::lang_start_internal\r\n  14: std::rt::lang_start\r\n             at /build/rust/src/rustc-1.36.0-src/src/libstd/rt.rs:64\r\n  15: main\r\n  16: __libc_start_main\r\n  17: _start\r\n\r\nwarning: build failed, waiting for other jobs to finish...\r\nerror: build failed\r\n```"
    },
    {
        "url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/21",
        "repository_url": "https://api.github.com/repos/rustbaeceans/discordterm",
        "labels_url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/21/labels{/name}",
        "comments_url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/21/comments",
        "events_url": "https://api.github.com/repos/rustbaeceans/discordterm/issues/21/events",
        "html_url": "https://github.com/rustbaeceans/discordterm/issues/21",
        "id": 324167567,
        "node_id": "MDU6SXNzdWUzMjQxNjc1Njc=",
        "number": 21,
        "title": "Not working",
        "user": {
            "login": "RealProva",
            "id": 26176059,
            "node_id": "MDQ6VXNlcjI2MTc2MDU5",
            "avatar_url": "https://avatars2.githubusercontent.com/u/26176059?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/RealProva",
            "html_url": "https://github.com/RealProva",
            "followers_url": "https://api.github.com/users/RealProva/followers",
            "following_url": "https://api.github.com/users/RealProva/following{/other_user}",
            "gists_url": "https://api.github.com/users/RealProva/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/RealProva/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/RealProva/subscriptions",
            "organizations_url": "https://api.github.com/users/RealProva/orgs",
            "repos_url": "https://api.github.com/users/RealProva/repos",
            "events_url": "https://api.github.com/users/RealProva/events{/privacy}",
            "received_events_url": "https://api.github.com/users/RealProva/received_events",
            "type": "User",
            "site_admin": false
        },
        "labels": [],
        "state": "open",
        "locked": false,
        "assignee": null,
        "assignees": [],
        "milestone": null,
        "comments": 0,
        "created_at": "2018-05-17T20:09:01Z",
        "updated_at": "2018-05-17T20:09:01Z",
        "closed_at": null,
        "author_association": "NONE",
        "body": "For some reason won't work, says failed to run \\ pkg-config no such file or dir "
    }
]